mypy/newsemanal/semanal.py

```diff
@@ -1877,14 +1881,57 @@ def add_type_alias_deps(self, aliases_used: Iterable[str],
             target = self.scope.current_target()
         self.cur_mod_node.alias_deps[target].update(aliases_used)
 
-    def is_not_ready_type_ref(self, rv: Expression) -> bool:
-        """Does this expression refers to a not-ready class?
+    def is_none_alias(self, node: Expression) -> bool:
+        """Is this a r.h.s. for a None alias?
 
-        This includes 'Ref' and 'Ref[Arg1, Arg2, ...]', where 'Ref'
-        refers to a PlaceholderNode with becomes_typeinfo=True.
+        We special case the assignments like Void = type(None), to allow using
+        Void in type annotations.
         """
-        if isinstance(rv, IndexExpr) and isinstance(rv.base, RefExpr):
-            return self.is_not_ready_type_ref(rv.base)
+        if isinstance(node, CallExpr):
+            if (isinstance(node.callee, NameExpr) and len(node.args) == 1 and
+                    isinstance(node.args[0], NameExpr)):
+                call = self.lookup_qualified(node.callee.name, node.callee)
+                arg = self.lookup_qualified(node.args[0].name, node.args[0])
+                if (call is not None and call.node and call.node.fullname() == 'builtins.type' and
+                        arg is not None and arg.node and arg.node.fullname() == 'builtins.None'):
+                    return True
+        return False
+
+    def is_type_ref(self, rv: Expression, bare: bool = False) -> bool:
+        """Does this expression refer to a type?
```

Comment:
The docstring doesn't make it clear whether `C[int]` would be considered a type reference. It would be good to be explicit about this. Also, probably worth mentioning that this assumes a type alias context. It's important since some of the error messages generated below only make sense when defining a type alias.